eng/scripts/job-matrix/README.md

```diff
@@ -202,19 +217,32 @@ The top level keys are used as job names, meaning they get displayed in the azur
 
 The logic for generating display names works like this:
 
-1. Sort the matrix by: parameter length, parameter name, and sort the parameter values arrays.
-2. In sorted order, join parameter values by "_"
+- Join parameter values by "_"
     a. If the parameter value exists as a key in `displayNames` in the matrix config, replace it with that value.
-    b. For each name value, strip all `.-_` characters from the string.
-
-The matrix is then sorted by display name, before being sent to azure pipelines. The underlying matrix may have a different
-sorted order than the display name output. The sorting needs to be separate so that sparse matrix generation can be deterministic.
+    b. For each name value, strip all non-alphanumeric characters (excluding "_").
+	c. If the name is greater than 100 characters, truncate it.
 
 #### Filters
 
-To be implemented. A basic example of a filter is "all" vs. "sparse" generation, but eventually will be a more programmable
-way of processing excludes, such as an expression that only includes entries with a container image specified. The intent
-is that these filters can be entered at runtime, as opposed to statically in yaml.
+Filters can be passed to the matrix as an array of strings, each matching one of three formats below. Multiple filters
+get treated with the `and` operator.
+
+1. `<key>=<regex>` - This will find a matrix parameter matching <key> and include it if the key exists and the value matches the regex filter.
+1. `?<key>=<regex>` - This will find a matrix parameter matching <key> and include it if the value matches the regex filter or the key does not exist.
```

Comment:
What is a scenario you expect we need this second filter case? It seems like it is is mostly here to work around the fact we don't have "or" support. I can see clear usages for the other 2 but this one isn't obvious to me.

Reply:
I'm using it right now in archetype-sdk-tests-generate.yml. We need to be able to filter for: if `SupportedClouds` is defined, it must contain the current cloud. If it's not defined then assume all clouds are supported. I think there will be other scenarios with similar logic like: opt-in or out if specified, otherwise default.

Reply:
Ok this one will definitely be the one I think people will need to keep referring back to the docs to understand what it means.

Reply:
Yeah perhaps instead of syntax having named arguments for different filters would be better

Reply:
Yes having independent options might be a little less magic.